packages/nx/bin/init-local.ts

```diff
@@ -229,3 +251,27 @@ function monkeyPatchRequire() {
     // do some side-effect of your own
   };
 }
+
+function shouldRestartWithExperimentalTsEsmLoader(): boolean {
+  // Already restarted with experimental loader
+  if (process.env.RESTARTED_WITH_EXPERIMENTAL_TS_ESM_LOADER === '1')
+    return false;
+  const nodeVersion = parseInt(process.versions.node.split('.')[0]);
+  // `--experimental-loader` is only supported in Nodejs >= 16 so there is no point restarting for older versions
+  if (nodeVersion < 16) return false;
+  // Node 20.6.0 adds `module.register`, otherwise we need to restart process with "--experimental-loader ts-node/esm".
+  return !require('node:module').register && moduleResolves('ts-node/esm');
+}
+
+function execArgvWithExperimentalLoaderOptions() {
+  return ['--no-warnings', '--experimental-loader', 'ts-node/esm'];
```

Comment:
We should probably account for options the user already passed in via `process.execArgv`